### DISCUSSION TITLE
Is mysql supported?

### DISCUSSION THREAD

**Discussion by oelbaga** [nextauthjs/next-auth#3727] [Help]
The faq section of the documentation says that mysql is a supported database for database sessions https://next-auth.js.org/faq#databases, although it is not listed in the adapter section. Is mysql really supported for database sessions?

**Comment by ghost** [comment]
I think only with Primsa or TypeORM. I would like one with mysql2. Maybe I'll create a custom adapter. Not sure if many people would use that?

**Reply by balazsorban44** [reply]
See https://github.com/nextauthjs/adapters/pull/347

We might accept one, but should be opened against this repository. Currently migrating #3650

**Reply by ghost** [reply]
I'm creating an adapter here:
https://github.com/roelandmoors/authjs-adapter-sql

The basic tests work, but needs some more testing.

**Reply by ghost** [reply]
published here for now: https://www.npmjs.com/package/authjs-adapter-sql

Let me know if you prefer a different location.

Uses plain sql and works with mysql2 and pg-promise

**Reply by ghost** [reply]
The [PlanetScale Serverless Driver](https://github.com/planetscale/database-js) is also supported

**Comment by clipartinc** [comment]
The NextAuth v4 F.A.Q. are an copied from v3 F.A.Q.s without any changes. I'm also wanting to use mySQL the best database option to use IMHO. Does nobody use mySQL anymore?

**Reply by oelbaga** [reply]
It blows my mind the lack of support MySQL gets in the full stack world while it's one of the most popular databases in the world. I had to write my own Auth for it. I'm not abandoning MySQL for a simple login authentication.

**Reply by clipartinc** [reply]
Yes, I agree!

So you ended up not using NextAuth at all?

**Reply by oelbaga** [reply]
NextAuth is good for simple login authentication that doesn't have to be tracked with a MySQL database. For projects where MySQL heavily involved and I'd like to have control of the server side cookie I write my own login authentication where the cookie id is also stored in the database.
